app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetWeightLifting.kt

```diff
@@ -0,0 +1,222 @@
+                isRepErrorVisible = true
+            }
+        },
+        repValue = repValue,
+        onRepValueChange = {
+            isRepErrorVisible = it.isBlank()
+            repValue = it
+        }
+    )
+    EditWorkoutMandatoryWeightSection(
+        isWeightErrorVisible = isWeightErrorVisible,
+        poundsOrKilogramsText = weightType.stringValue,
+        weightValue = weightValue,
+        onWeightTypeClicked = {
+            weightType = if (weightType == WeightLiftingWeightType.POUNDS) {
+                WeightLiftingWeightType.KILOGRAMS
+            } else {
+                WeightLiftingWeightType.POUNDS
+            }
+        },
+        onWeightValueChange = {
+            isWeightErrorVisible = it.isBlank()
+            weightValue = it
+        },
+        editWeightValue = { editWorkoutFunction: EditWorkoutFunction, weight: String ->
+            isWeightErrorVisible = false
+            try {
+                when {
+                    weight.isEmpty() || weight.toDoubleOrZero() == 0.0 -> {
+                        weightValue = when (editWorkoutFunction) {
+                            EditWorkoutFunction.ADD_VALUE -> "1.0"
+                            EditWorkoutFunction.SUBTRACT_VALUE -> "0"
+                        }
+                    }
+
+                    weight.last() == '.' -> {
+                        val oldWeight = weight.substringBefore(".")
+                        weightValue = try {
+                            when (editWorkoutFunction) {
+                                EditWorkoutFunction.ADD_VALUE -> (oldWeight.toDoubleOrZero() + 1).toString()
+                                EditWorkoutFunction.SUBTRACT_VALUE -> (oldWeight.toDoubleOrZero() - 1).toString()
+                            }
+                        } catch (e: Exception) {
+                            when (editWorkoutFunction) {
+                                EditWorkoutFunction.ADD_VALUE -> "1.0"
+                                EditWorkoutFunction.SUBTRACT_VALUE -> "0"
+                            }
+                        }
+                    }
+
+                    else -> {
+                        weightValue =
+                            when (editWorkoutFunction) {
+                                EditWorkoutFunction.ADD_VALUE -> (weight.toDoubleOrZero() + 1).toString()
+                                EditWorkoutFunction.SUBTRACT_VALUE -> {
+                                    val newWeight = weight.toDoubleOrZero() - 1
+                                    when {
+                                        newWeight <= 0 -> "0"
+                                        else -> newWeight.roundToTwoDecimalPlaces().toString()
+                                    }
+                                }
+                            }
+                    }
+                }
+                if (weightValue.toDoubleOrNull() == null) {
+                    isWeightErrorVisible = true
+                }
```

Comment:
This entire callback function is identical to the the EditWorkoutOptionalWeightSection Composable inside EditWorkoutSetCalisthenics.kt except for lines 206-209 on this file. Is this check something that was missed in the EditWorkoutSetCalisthenics file? If so, would this be extracted into a helper function and be reused?